app/src/main/java/com/duckduckgo/app/httpsupgrade/db/HTTPSUpgradeDomain.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2017 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.httpsupgrade.db
+
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Ignore
+import android.arch.persistence.room.PrimaryKey
+
+@Entity(tableName = "https_upgrade_domain")
+class HTTPSUpgradeDomain {
+
+    @PrimaryKey var domain: String = ""
+
+    // Required for Room
+    @Suppress("unused")
+    constructor()
```

Comment:
Could this be removed by adding the default constructor to the class definition? and then on the secondary constructor, delegating to the primary one. eg, 
 
    class HTTPSUpgradeDomain() {
    
        @PrimaryKey var domain: String = ""
    
        @Ignore
        constructor(domain: String) : this() {
            this.domain = domain
        }
    }